crates/sui-node/src/lib.rs

```diff
@@ -83,7 +83,13 @@ impl SuiNode {
             .await,
         );
 
-        let gossip_handle = if config.enable_gossip {
+        // TODO: maybe have a config enum that takes care of this for us.
+        let is_validator = config.consensus_config().is_some();
```

Comment:
We have separate configs for fullnode (constructed in swarm.rs) and for validators (builder.rs), so enable gossip can be set respectively and we don't need this hack?

Reply:
This isn't a hack, there is only a single config `NodeConfig` and it is used for the sui-node binary (which is the same binary used for validators and fullnodes). Somethings are going to need to be tweaked depending on if we're running in fullnode or validator mode.

Reply:
We probably could make this an enum that is queryable from the config itself and the enum could be:
```rust
enum Role {
    Validator,
    Fullnode,
}
```